### PR TITLE
feat: implemented CustomGlobalExceptionHandler and added meaningful v…

### PR DESCRIPTION
**Pull request by PavloMelnyk10**
…alidation annotations to DTO classes for improved error handling and data validation.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -80,6 +80,12 @@
             <artifactId>liquibase-maven-plugin</artifactId>
             <version>${liquibase.version}</version>
         </dependency>
+        <!-- Hibernate validator dependency -->
+        <dependency>
+            <groupId>org.hibernate</groupId>
+            <artifactId>hibernate-validator</artifactId>
```
Comment (liliia-ponomarenko) [inline pom.xml]: better use spring-boot-starter-validation

File: src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -8,10 +12,28 @@
 @Setter
 public class CreateBookRequestDto {
     private Long id;
+
+    @NotNull(message = "Please provide a book title")
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java]: with literals better using NotBlank

File: src/main/java/mate/academy/bookstore/dto/UpdateBookRequestDto.java [hunk 3]
```diff
@@ -1,16 +1,37 @@
...
+import jakarta.validation.constraints.Pattern;
+import jakarta.validation.constraints.Positive;
+import jakarta.validation.constraints.Size;
 import java.math.BigDecimal;
 import lombok.Getter;
 import lombok.Setter;
 
 @Getter
 @Setter
 public class UpdateBookRequestDto {
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/bookstore/dto/UpdateBookRequestDto.java:14]: the same

File: src/main/java/mate/academy/bookstore/dto/UpdateBookRequestDto.java [hunk 4]
```diff
@@ -1,16 +1,37 @@
...
+    @Size(min = 3, max = 100, message = "The author's name must be between 3 and 100 characters")
     private String author;
+
+    @NotNull(message = "ISBN can't be null during update")
+    @Pattern(regexp = "^(978|979)-\\d{10}$",
+            message = "Invalid ISBN format. It should be like 978-1234567891")
     private String isbn;
+
+    @NotNull(message = "Price is required")
+    @Positive(message = "Price must be greater than zero")
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/bookstore/dto/UpdateBookRequestDto.java:31]: if I remember correctly Positive includes a NotNull check

File: src/main/java/mate/academy/bookstore/exception/CustomGlobalExceptionHandler.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    @ExceptionHandler(EntityNotFoundException.class)
+    public ResponseEntity<Object> handleEntityNotFound(
+            EntityNotFoundException ex, WebRequest request) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.NOT_FOUND.value());
+        body.put("error", "Entity not found");
+        body.put("message", ex.getMessage());
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/bookstore/exception/CustomGlobalExceptionHandler.java]: you can move it to the private method and reuse it ;)

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s  improve your solution ;)

File: src/main/java/mate/academy/bookstore/dto/UpdateBookRequestDto.java (hunk 4 above)
Comment (PavloMelnyk10) [reply src/main/java/mate/academy/bookstore/dto/UpdateBookRequestDto.java:31]: I've checked and Positive check doesn't include a Null check

File: src/main/java/mate/academy/bookstore/exception/CustomGlobalExceptionHandler.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+    private String getErrorMessage(ObjectError e) {
+        if (e instanceof FieldError) {
+            String fieldName = ((FieldError) e).getField();
+            String message = e.getDefaultMessage();
+            return fieldName + ": " + message;
+        }
+        return e.getDefaultMessage();
+    }
+
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/bookstore/exception/CustomGlobalExceptionHandler.java:49]: move it after public methods

**Review by liliia-ponomarenko — APPROVED** [review]
Great! 👍

**Pull request merged**